multichain-testing/test/ibc-transfers.ts

```diff
@@ -0,0 +1,291 @@
+import anyTest from '@endo/ses-ava/prepare-endo.js';
+import { execFileSync } from 'node:child_process';
+import type { TestFn, ExecutionContext } from 'ava';
+import { commonSetup, type SetupContext } from './support.js';
+import { createWallet, generateMnemonic } from '../tools/wallet.js';
+import { makeQueryClient } from '../tools/query.js';
+import { makeAgd, type Agd } from '../tools/agd-lib.js';
+import starshipChainInfo from '../starship-chain-info.js';
+import type { CosmosChainAddress, ForwardInfo } from '@agoric/orchestration';
+import { sleep } from '../tools/sleep.js';
+import { objectMap } from '@endo/patterns';
+
+const test = anyTest as TestFn<SetupContext>;
+
+test.before(async t => {
+  t.context = await commonSetup(t);
+});
+
+const txEventQuery = /** @type {const} */ {
+  msgReceivePacket: {
+    event: 'message.action',
+    value: '/ibc.core.channel.v1.MsgRecvPacket',
+  },
+  msgAcknowledgement: {
+    event: 'message.action',
+    value: '/ibc.core.channel.v1.MsgAcknowledgement',
+  },
+};
+
+// use this on osmosis, cosmoshub as our account name in the keyring
+const keyName = 'testuser';
+
+const fundRemote = async (
+  t: ExecutionContext<SetupContext>,
+  {
+    srcChainName,
+    destChainName,
+    acctAddr,
+    qty,
+    denom,
+  }: {
+    srcChainName: string;
+    destChainName: string;
+    acctAddr: string;
+    qty: number;
+    denom: string;
+  },
+) => {
+  const { agd, retryUntilCondition, useChain } = t.context;
+  const destChainid = useChain(destChainName).chain.chain_id;
+  const srcChainId = useChain(srcChainName).chain.chain_id;
+
+  const agoricToOsmosis =
+    starshipChainInfo[srcChainName].connections[destChainid].transferChannel;
+  t.assert(agoricToOsmosis.channelId);
+  const amount = `${qty}${denom}`;
+  // agd tx ibc-transfer transfer [src-port] [src-channel] [receiver] [amount] [flags]
+  const fundInitial = await agd.tx(
+    [
+      'ibc-transfer',
+      'transfer',
+      'transfer',
+      agoricToOsmosis.channelId,
+      acctAddr,
+      amount,
+    ],
+    {
+      chainId: srcChainId,
+      from: 'faucet',
+      yes: true,
+    },
+  );
+  t.is(fundInitial?.code, 0);
+  t.log('Success. TxHash:', fundInitial?.txhash);
+
+  const queryClient = makeQueryClient(
+    await useChain(destChainName).getRestEndpoint(),
+  );
+  const { balances } = await retryUntilCondition(
+    () => queryClient.queryBalances(acctAddr),
+    ({ balances }) => !!balances.find(x => x.denom.startsWith('ibc/')),
+    `${acctAddr} received bld from agoric`,
+  );
+
+  t.log('amtOnDest', `${qty}${balances[0].denom}`);
+  return { denom: balances[0].denom };
+};
+
+const setupSourceWallet = async (
+  t: ExecutionContext<SetupContext>,
+  { chainName }: { chainName: string },
+) => {
+  const { retryUntilCondition, useChain } = t.context;
+
+  const chaind = makeAgd({ execFileSync }).withOpts({
```

Comment:
good name. let's use it for the lib?
```suggestion
  const chaind = makeChaind({ execFileSync }).withOpts({
```

Reply:
Good suggestion. Technically now `makeChaind` should then take `chainName` as argument so that it returns relevant `chaind` using a map under-the-hood.

Reply:
It already takes `chainName` and can be used with `withOpts`. I'll change

Reply:
I'll refactor this in a separate github issue:
https://github.com/Agoric/agoric-sdk/issues/11496